core/src/main/java/com/netease/arctic/data/FileNameRules.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.data;
+
+import com.netease.arctic.ams.api.Constants;
+import com.netease.arctic.io.writer.TaskWriterKey;
+import com.netease.arctic.utils.IdGenerator;
+import com.netease.arctic.utils.TableFileUtils;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * File name pattern:${tree_node_id}-${file_type}-${transaction_id}-${partition_id}-${task_id}-{operation_id}-{count}
+ * <ul>
+ *   <li>tree_node_id: id of {@link DataTreeNode} the file belong</li>
+ *   <li>file_type: short name of file's {@link DataFileType} </li>
+ *   <li>transaction_id: id of transaction the file added</li>
+ *   <li>partition_id: id of partitioned data in parallel engine like spark & flink </li>
+ *   <li>task_id: id of write task within partition</li>
+ *   <li>operation_id: a random id to avoid duplicated file name</li>
+ *   <li>count: auto increment count within writer </li>
+ * </ul>
+ * like:
+ * <ul>
+ *   <li>1-B-100-0-0-4217271085623029157-0.parquet</li>
+ *   <li>4-ED-101-0-0-9009257362994691056-1.parquet</li>
+ *   <li>4-I-101-0-0-9009257362994691056-2.parquet</li>
+ * </ul>
+ * 
+ */
+public class FileNameRules {
+
+  private static final String KEYED_FILE_NAME_PATTERN_STRING = "(\\d+)-(\\w+)-(\\d+)-(\\d+)-(\\d+)-.*";
+  private static final Pattern KEYED_FILE_NAME_PATTERN = Pattern.compile(KEYED_FILE_NAME_PATTERN_STRING);
```

Comment:
This class is not introduced from 0.5, It has another name in 0.4,  do not add this as new file.